feature: add OLX Example course, for now a copy of edX-Insider

pdpinch opened the pull request:
This is the first step in addressing the docs issue https://github.com/openedx/docs.openedx.org/issues/998

> Solution: Create a toy OLX course within https://github.com/openedx/training-courses for people to browse, and update the instructions in this page tree to match.

This PR contains a simple copy of the edX-Insider course. In a follow-up PR, we could simplify the course to focus solely on the content needed to support OLX documentation. 

I followed the existing pattern in this repo of commiting tar.gz files. However, to support the OLX documentation as currently written, it would be convenient to have an exploded copy of the repo, so individual files can be viewed. Would that be acceptable in this repo?

openedx-webhooks commented:
Thanks for the pull request, @pdpinch!


This repository is currently maintained by `@sarina`.


Once you've gone through the following steps feel free to tag them in a comment and let them know that your changes are ready for engineering review.

<details><summary>:radio_button: Get product approval</summary>

If you haven't already, [check this list](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Does-my-contribution-require-Product-Review%3F) to see if your contribution needs to go through the product review process.

- If it does, you'll need to submit a product proposal for your contribution, and have it reviewed by the [Product Working Group](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3449028609/Product+Working+Group).
    - This process (including the steps you'll need to take) is documented [here](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Product-Review-Process).
- If it doesn't, simply proceed with the next step.
</details>

<details><summary>:radio_button: Provide context</summary>

To help your reviewers and other members of the community understand the purpose and larger context of your changes, feel free to add as much of the following information to the PR description as you can:

- Dependencies
  > This PR must be merged before / after / at the same time as ...
- Blockers
  > This PR is waiting for OEP-1234 to be accepted.
- Timeline information
  > This PR must be merged by XX date **because** ...
- Partner information
  > This is for a course on edx.org.
- Supporting documentation
- Relevant [Open edX discussion forum](https://discuss.openedx.org/) threads
</details>





<details><summary>:radio_button: Get a green build</summary>

If one or more checks are failing, continue working on your changes until this is no longer the case and your build turns green.
</details><details>


---

<details><summary>Where can I find more information?</summary>

If you'd like to get more details on all aspects of the review process for open source pull requests (OSPRs), check out the following resources:

- [Overview of Review Process for Community Contributions](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/FAQ-about-pull-requests.html)
- [Pull Request Status Guide](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/pull-request-statuses.html)
- [Making changes to your pull request](https://docs.openedx.org/en/latest/documentors/how-tos/make_changes_to_your_pull_request.html)
</details>

<details><summary>When can I expect my changes to be merged?</summary>

Our goal is to get community contributions seen and reviewed as efficiently as possible.

However, the amount of time that it takes to review and merge a PR can vary significantly based on factors such as:

- The size and impact of the changes that it introduces
- The need for product review
- Maintenance status of the parent repository

:bulb: *As a result it may take up to several weeks or months to complete a review and merge your PR.*
</details>
<!-- comment:external_pr -->
<!-- data: eyJkcmFmdCI6IGZhbHNlfQ== -->

sarina commented:
> I followed the existing pattern in this repo of commiting tar.gz files. However, to support the OLX documentation as currently written, it would be convenient to have an exploded copy of the repo, so individual files can be viewed. Would that be acceptable in this repo?

Yes. Actually I've been meaning to expand out all the courses in this repo so the text can be inspected/searched, but I haven't gotten around to it.

sarina commented:
It would be good to follow the pattern of https://github.com/openedx/openedx-demo-course where there's both the unpacked course and a downloadable one.

pdpinch commented:
Does this work? I added a `course` directory with the uncompressed course files and moved the .tar.gz file to a `dist` directory -- kind of like the directory structure of https://github.com/openedx/openedx-demo-course but not exactly. 

I'm a little confused by this repo having multiple courses and the other one having just one.

sarina commented:
I would structure this like:

OLX Example Course
|-> dist/
|-> olx_example_course/
|-> |-> course/
|-> |-> README.rst

as well as update the top-level README on this repo

sarina commented:
I would also make a follow-on ticket in this repo to update the course to be Open edX specific

sarina commented:
@pdpinch - I disagree a bit with my previous comment. I inspected the demo-course repo and the structure there is actually exactly what we want (they have 1 course and 1 library, whereas we'll have at least 4 courses). Check out what I added in #62

sarina commented:
#65 is a cleaner example of just adding a new course to this repo.
